mesonbuild/interpreter/compiler.py

```diff
@@ -448,9 +448,12 @@ def get_define_method(self, args: T.Tuple[str], kwargs: 'CommonKW') -> str:
     def compiles_method(self, args: T.Tuple['mesonlib.FileOrString'], kwargs: 'CompileKW') -> bool:
         code = args[0]
         if isinstance(code, mesonlib.File):
+            if code.is_built:
+                FeatureNew.single_use('compiler.compiles with built File object', '1.2.0', self.subproject,
```

Comment:
As a user I would be very puzzled by what a "built File object" is. Is there a way we can reword this to be a little easier to grok?

Reply:
I agree it's a bit cryptic message. Maybe "with the return value of configure_file()" as I don't think we have other ways to get a built file? Or "with a file created at configure time"?

Also, the message in links_method() is different, shouldn't we have the same message in both cases?

Reply:
> Also, the message in links_method() adds "It was silently broken and always returned false.", shouldn't we have that here too?

Doesn't it already?

Reply:
> Doesn't it already?

Oh yes, misread because this comment thread in between both lines, sorry.